src/TurtleParserMain.cpp

```diff
@@ -2,36 +2,166 @@
 // Chair of Algorithms and Data Structures.
 // Author: Johannes Kalmbach(joka921) <johannes.kalmbach@gmail.com>
 
+#include <getopt.h>
 #include <array>
 #include <fstream>
 #include <iostream>
 #include <string>
+#include "./parser/NTriplesParser.h"
+#include "./parser/TsvParser.h"
 #include "./parser/TurtleParser.h"
+#include "./util/Log.h"
 
-void writeNT(std::ostream& out, const std::string& filename) {
-  TurtleParser p(filename);
+/**
+ * @brief Instantiate a Parser that parses filename and writes the resulting
+ * triples to argument out.
+ *
+ * @tparam Parser A Parser that supports a call to getline that yields a triple
+ * @param out the parsed triples are written to this file
+ * @param filename the filename from which the triples are parsed, can be
+ * "/dev/stdin"
+ */
+template <class Parser>
+void writeNTImpl(std::ostream& out, const std::string& filename) {
+  Parser p(filename);
   std::array<std::string, 3> triple;
-  while (p.getLine(&triple)) {
+  // this call by reference is necesary because of the TSV-Parsers interface
+  while (p.getLine(triple)) {
     out << triple[0] << " " << triple[1] << " " << triple[2] << " .\n";
   }
 }
 
-int main(int argc, char** argv) {
-  if (argc < 2 || argc > 3) {
-    std::cerr << "Usage: ./TurtleParserMain <turtleInput> [<nTripleOutput>]"
-              << std::endl;
+/**
+ * @brief Decide according to arg fileFormat which parser to use.
+ * Then call writeNTImpl with the appropriate parser
+ * @param out Parsed triples will be written here.
+ * @param fileFormat One of [tsv|bzip2|ttl|tsv]
+ * @param filename Will read from this file, might be /dev/stdin
+ */
+void writeNT(std::ostream& out, const string& fileFormat,
+             const std::string& filename) {
+  if (fileFormat == "ttl") {
+    writeNTImpl<TurtleStreamParser>(out, filename);
+  } else if (fileFormat == "tsv") {
+    writeNTImpl<TsvParser>(out, filename);
+  } else if (fileFormat == "nt") {
+    writeNTImpl<NTriplesParser>(out, filename);
+  } else if (fileFormat == "mmap") {
+    writeNTImpl<TurtleMmapParser>(out, filename);
+  } else {
+    LOG(ERROR) << "writeNT was called with unknown file format " << fileFormat
+               << ". This should never happen, terminating" << std::endl;
+    LOG(ERROR) << "Please specify a valid file format" << std::endl;
     exit(1);
   }
+}
+
+// _______________________________________________________________________________________________________________
+void printUsage(char* execName) {
+  std::ios coutState(nullptr);
+  coutState.copyfmt(cout);
+  cout << std::setfill(' ') << std::left;
+
+  cout << "  " << std::setw(20) << "F, file-format" << std::setw(1) << "    "
+       << " Specify format of the input file. Must be one of "
+          "[tsv|nt|ttl|mmap]."
```

Comment:
You mention `mmap` here but later check for `bzip2` not `mmap`.

Reply:
Fixed